### DISCUSSION TITLE
Edit dialog from edit

### DISCUSSION THREAD

**Discussion by Dev-Lopez02** [serenity-is/Serenity#7009] [Help]
I am trying to activate the edit mode of another dialog but when I click on it, it only opens to add a new dialog. 

	else if (target.hasClass('new-order')) {
					var contratoDialog = new CFNCable.Clientes.ContratoDialog();
					this.initDialog(contratoDialog);

					// Crear un objeto que contenga el valor de Idecnt
					var idecntValue = item.Idecnt;

					// Abre el diálogo en modo de edición y pasa el valor de Idecnt
					contratoDialog.editItem(idecntValue);
				}

///////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////

namespace CFNCable.Clientes {

    @Serenity.Decorators.registerClass()
    export class DetContratiGrid extends Serenity.EntityGrid<DetContratiRow, any> {
		protected getColumnsKey() { return "Clientes.DetContrati"; }
		protected getDialogType() { return <any> DetContratiDialog; }
        protected getIdProperty() { return DetContratiRow.idProperty; }
        //protected getInsertPermission() { return DetContratiRow.insertPermission; }
        protected getLocalTextPrefix() { return DetContratiRow.localTextPrefix; }
        protected getService() { return DetContratiService.baseUrl; }

        constructor(container: JQuery) {
            super(container);
		}

		protected getButtons(): Serenity.ToolButton[] {

			// call base method to get list of buttons
			// by default, base entity grid adds a few buttons, 
			// add, refresh, column selection in order.
			var buttons = super.getButtons();

			// here is several methods to remove add button
			// only one is enabled, others are commented

			// METHOD 1
			// we would be able to simply return an empty button list,
			// but this would also remove all other buttons
			// return [];

			// METHOD 2
			// remove by splicing (something like delete by index)
			// here we hard code add button index (not nice!)
			// buttons.splice(0, 1);

			// METHOD 3 - recommended
			// remove by splicing, but this time find button index
			// by its css class. it is the best and safer method
			buttons.splice(Q.indexOf(buttons, x => x.cssClass == "add-button"), 1);

			return buttons;
		}

		protected getColumns() {
			var columns = super.getColumns();

			columns.unshift({
				field: 'Delete Row',
				name: '',
				format: ctx => '<a class="inline-action delete-row" title="delete">' +
					'<i class="fa fa-trash-o text-red"></i></a>',
				width: 24,
				minWidth: 24,
				maxWidth: 24
			});

			columns.splice(1, 0, {
				field: 'Ver Cliente',
				name: '',
				format: ctx => '<a class="inline-action view-details" title="ver cliente"></a>',
				width: 24,
				minWidth: 24,
				maxWidth: 24
			});

			columns.splice(2, 0, {
				field: 'Ver Contrato',
				name: '',
				format: ctx => '<a class="inline-action new-order" title="ver contrato"></a>',
				width: 24,
				minWidth: 24,
				maxWidth: 24
			});

			return columns;
		}

		protected onClick(e: JQueryEventObject, row: number, cell: number) {
			super.onClick(e, row, cell);

			if (e.isDefaultPrevented())
				return;

			var item = this.itemAt(row);
			var target = $(e.target);

			// if user clicks "i" element, e.g. icon
			if (target.parent().hasClass('inline-action'))
				target = target.parent();

			if (target.hasClass('inline-action')) {
				e.preventDefault();

				if (target.hasClass('delete-row')) {
					Q.confirm('Delete record?', () => {
						Clientes.ClientesService.Delete({
							EntityId: item.IdecntIdecli,
						}, response => {
							this.refresh();
						});
					});
				}
				else if (target.hasClass('new-order')) {
					var contratoDialog = new CFNCable.Clientes.ContratoDialog();
					this.initDialog(contratoDialog);

					// Crear un objeto que contenga el valor de Idecnt
					var idecntValue = item.Idecnt;

					// Abre el diálogo en modo de edición y pasa el valor de Idecnt
					contratoDialog.editItem(idecntValue);
				}

				else if (target.hasClass('view-details')) {
					var dlg = new Clientes.ClientesDialog();
					this.initDialog(dlg);
					dlg.loadEntityAndOpenDialog(<Clientes.ContratoRow>{
						Idecli: item.IdecntIdecli
					});
				}
			}
		}
    }
}

**Comment by luongnv6** [comment]
you just need use  loadEntityAndOpenDialog or loadByIdAndOpenDialog
